File: Programmers Level1/체육복.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <math.h>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
	vector<int> temp(n);
	for (int i = 0; i < n; i++) {
		temp[i]++;
	}
	for (int i = 0; i < reserve.size(); i++) {
		temp[reserve[i] - 1]++;
	}
	for (int i = 0; i < lost.size(); i++) {
		temp[lost[i] - 1]--;
	}
	for (int i = 0; i < temp.size(); i++) {
		if (temp[i] == 0 && (temp[i - 1] > 1)) {
			temp[i]++; temp[i - 1]--;
		}

		else if (temp[i] == 0 && (temp[i + 1] > 1))
		{
			if (!((i + 1) == temp.size())) {
				temp[i]++; temp[i + 1]--;
			}
		}
	}

	int count = 0;
	for (int i = 0; i < temp.size(); i++) {
		if (temp[i] >= 1) {
			count++;
		}
	}
	return count;
}

File: Programmers Level1 - Kotlin/체육복.kt
fun main() {
    solution(5, intArrayOf(1,2,3), intArrayOf(2,3,4))
}
fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
    var answer = 0
    var temp = IntArray(n) {1}
    for (element in reserve) {
        temp[element - 1]++;
    }
    for (element in lost) {
        temp[element - 1]--;
    }
    for (i in temp.indices) {
        if (temp[i] == 0 && i != 0 &&(temp[i - 1] > 1)) {
            temp[i]++; temp[i - 1]--;
        }

        else if (temp[i] == 0 && i != (temp.size-1) &&(temp[i + 1] > 1))
        {
            if ((i + 1) != temp.size) {
                temp[i]++; temp[i + 1]--;
            }
        }
    }


    for (element in temp) {
        if (element >= 1) {
            answer++;
        }
    }
    println(answer)
    return answer
}